api/base/filters.py

```diff
@@ -29,15 +31,49 @@ def query_params_to_fields(query_params):
 def intersect(x, y):
     return x & y
 
+
 class ODMFilterMixin(object):
     """View mixin that adds a get_query_from_request method which converts query params
     of the form `filter[field_name]=value` into an ODM Query object.
 
     Subclasses must define `get_default_odm_query()`.
+
+    Serializers that want to restrict which fields are used for filtering need to have a variable called
+    filterable_fields which is a list of strings representing the field names as they appear in the serialization.
     """
+
+    # TODO Handle simple and complex non-standard fields
+
     TRUTHY = set(['true', 'True', 1, '1'])
     FALSY = set(['false', 'False', 0, '0'])
 
+    # For the field_comparison_operators, instances can be a class or a tuple of classes
+    field_comparison_operators = [
+        {
+            'instances': ser.CharField,
+            'comparison_operator': 'icontains'
+        },
+        {
+            'instances': ser.ListField,
+            'comparison_operator': 'in'
+        }
+    ]
+
+    def get_comparison_operator(self, key, value):
+        default_operator = 'eq'
+
+        for operator in self.field_comparison_operators:
+            if isinstance(self.serializer_class._declared_fields[key], operator['instances']):
+                return operator['comparison_operator']
+
+        return default_operator
+
+    def is_filterable_field(self, key, value):
```

Comment:
Yeah, I normally wouldn't have added that in, but you had something that was essentially the same, so I wasn't sure if you had just left it in or wanted some sort of symmetry between the different parts of the Q function from line 92. I'm happy to remove, though, in each of the methods that aren't using all of the values.